### PR TITLE
ios: replace Diawi Fastlane plugin that disappeared

### PR DESCRIPTION
**Pull request by jakubgs**
For an unknown reason the original Diawi plugin for Fastlane has been removed from GitHub and RubyGems pages and can no longer be used. This replaces it with a Node.js script which does the same job.
    
I tried using `diawi` and `diawi-nodejs-uploader` but both had issues, one of them being depending on far too many useless packages.

Also, I hate the JavaScript callback hell so much. And even if you try to use `async`/`await` you still have to deal with awkward custom `Promise` definitions that are not great for debugging or just clarity in general.

Depends on:
* https://github.com/status-im/status-jenkins-lib/pull/69

Resolves:
* https://github.com/status-im/status-mobile/issues/15951

### CODE REVIEW

**Comment by jakubgs** [comment]
It's failing with:
```
11:11:51  Uploading: StatusIm-Mobile-230519-090551-91d437-pr15959.ipa
11:11:53  { error: 4000000, message: 'Too many requests' }
```
Weird.

**Review by Parveshdhull — APPROVED** [review]

**Review by ilmotta — APPROVED** [review]

**Review by smohamedjavid — APPROVED** [review]

**Comment by jakubgs** [comment]
It's weird, but it seem like the `Too many requests` error shows up only on the new Mac M2 host, which is very weird:

* https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-15959/22/
* https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-15959/24/

But most of the time it works, not sure what's the difference. I keep trying to reach Diawi support.

**Comment by jakubgs** [comment]
I'm asking for a re-review since I replaced the previous solution with a custom script to avoid adding useless dependencies.

**Comment by jakubgs** [comment]
Also, I wonder if we need the `fastlane-plugin-clean_testflight_testers` module for anything. I don't think we do.

**Review by Parveshdhull — APPROVED** [review]

File: scripts/diawi-upload.mjs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+          message: res.statusMessage,
+          payload: payload,
+        })
+      })
+    })
+  })
+}
+
+const uploadIpa = async (ipaPath, comment, token) => {
+  var form = new FormData()
```
Comment (siddarthkay) [inline scripts/diawi-upload.mjs]: Tiny nit pick over here :) 
It is better to use `const` or `let` for variable declarations that have block scope.

**Review by siddarthkay — APPROVED** [review]
GG 🚀

File: scripts/diawi-upload.mjs (hunk 1 above)
Comment (jakubgs) [reply scripts/diawi-upload.mjs]: Ah yeah, must have copy pasted that.

**Review by smohamedjavid — APPROVED** [review]
LGTM 👍

**Comment by smohamedjavid** [comment]
> Also, I wonder if we need the `fastlane-plugin-clean_testflight_testers` module for anything. I don't think we do.

The QA team are only users in Testflight. I think we don't need it (until we add actual users in Testflight).

**Pull request merged**